src/Configuration/test/CloudFoundry.Test/CloudfoundryConfigurationProviderTest.cs

```diff
@@ -216,6 +224,101 @@ public void Load_VCAP_APPLICATION_Allows_Reload_Without_Throwing_Exception()
         Assert.Equal("fb8fbcc6-8d58-479e-bcc7-3b4ce5a7f0ca", options.Version);
     }
 
+    [Fact]
+    public async Task ForwardedHeaders_enabled_on_CloudFoundry()
+    {
+        using var vcapScope = new EnvironmentVariableScope("VCAP_APPLICATION", "{}");
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        await using WebApplication host = builder.Build();
+
+        ForwardedHeadersOptions options = host.Services.GetRequiredService<IOptions<ForwardedHeadersOptions>>().Value;
+
+        options.ForwardedHeaders.Should().HaveFlag(ForwardedHeaders.XForwardedFor);
+        options.ForwardedHeaders.Should().HaveFlag(ForwardedHeaders.XForwardedProto);
+        options.KnownNetworks.Should().BeEmpty();
+        options.KnownProxies.Should().BeEmpty();
+    }
+
+    [Fact]
+    public async Task ForwardedHeaders_not_enabled_off_CloudFoundry()
+    {
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        await using WebApplication host = builder.Build();
+
+        ForwardedHeadersOptions options = host.Services.GetRequiredService<IOptions<ForwardedHeadersOptions>>().Value;
+
+        options.ForwardedHeaders.Should().Be(ForwardedHeaders.None);
+        options.KnownNetworks.Should().ContainSingle().Which.Should().BeEquivalentTo(IPNetwork.Parse("127.0.0.1/8"));
+        options.KnownProxies.Should().ContainSingle().Which.Should().Be(IPAddress.Parse("::1"));
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task ForwardedHeadersMiddleware_enabled_as_expected(bool onCloudFoundry)
+    {
+        using var vcapScope = new EnvironmentVariableScope("VCAP_APPLICATION", onCloudFoundry ? "{}" : null);
+
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        await using WebApplication host = builder.Build();
+        bool? isHttpRequest = null;
+
+        host.Map("/", context =>
+        {
+            isHttpRequest = context.Request.IsHttps;
+            return Task.CompletedTask;
+        });
+
+        await host.StartAsync(TestContext.Current.CancellationToken);
+        TestServer server = host.GetTestServer();
+
+        HttpContext context = await server.SendAsync(c =>
```

Comment:
Please don't use single-character variable names.

`context` should be renamed to `httpContext`, we also have `TestContext` here.